.github/workflows/microbenchmark.yml

```diff
@@ -0,0 +1,52 @@
+name: microbenchmark
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+    tags:
+      - "v*.*.*"
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+env:
+  GO_VERSION: 1.15.10
```

Comment:
**thought**: I wonder if we should run the benchmark on the latest go version to get more "accurate" results

Reply:
No strong opinion, I literally migrated the existing implementation. If changed, what version? I don't know whether an annotation will be possible so in the Kibana dashboard there will be a explicit information about when the version changed.

Reply:
We can probably open a followup issue to discuss it